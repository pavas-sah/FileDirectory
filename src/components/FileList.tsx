import React, { Component } from 'react';
import data from '../data/data.json';
import FileListItem from './FileListItem';
import './FileList.css';

export interface FileState {
  id?: number;
  fileIcon: string;
  fileIconAlt: string;
  fileName: string;
  message: string;
  duration: string;
}

export class FileList extends Component {
  fileList = data.map((item: FileState) => {
    return (
      <span className='collection-item' key={item.id}>
        <FileListItem
          fileIcon={require(`../icons/${item.fileIcon}.png`)}
          fileIconAlt={item.fileIconAlt}
          fileName={item.fileName}
          message={item.message}
          duration={item.duration}
        />
      </span>
    );
  });

  render() {
    return <>{this.fileList}</>;
  }
}

export default FileList;
