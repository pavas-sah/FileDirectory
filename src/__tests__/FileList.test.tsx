import React from 'react';
import FileList, { FileState } from '../components/FileList';
import renderer from 'react-test-renderer';

const mockDuration = '12-5-2010';
const mockIcon = 'iconLocation';
const mockAlt = 'iconAlt';
const mockFileName = 'file';
const mockMessage = 'message';
const mockId = 1;

const fileProps: FileState = {
  duration: mockDuration,
  fileIcon: mockIcon,
  fileIconAlt: mockAlt,
  fileName: mockFileName,
  message: mockMessage,
  id: mockId,
};

describe('File List Page', () => {
  it('renders correctly', () => {
    const fileListComponent = renderer.create(<FileList {...fileProps} />);
    expect(fileListComponent.getInstance()?.props).toStrictEqual(fileProps);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<FileList {...fileProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
