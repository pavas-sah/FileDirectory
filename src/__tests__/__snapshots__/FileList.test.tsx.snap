// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File List Page matches snapshot 1`] = `
Array [
  <span
    className="collection-item"
  >
    <div
      className="sc-AxjAm jIgIEX"
    >
      <img
        alt="Html"
        src={Object {}}
      />
      Html
      Learn HTML For Beginners With Free Book 
      19 years ago
    </div>
  </span>,
  <span
    className="collection-item"
  >
    <div
      className="sc-AxjAm jIgIEX"
    >
      <img
        alt="Html"
        src={Object {}}
      />
      CSS
      Learn CSS For Beginners With Free Book 
      23 years ago
    </div>
  </span>,
  <span
    className="collection-item"
  >
    <div
      className="sc-AxjAm jIgIEX"
    >
      <img
        alt="Html"
        src={Object {}}
      />
      Html
      Learn JS For Beginners With Free Book 
      10 years ago
    </div>
  </span>,
  <span
    className="collection-item"
  >
    <div
      className="sc-AxjAm jIgIEX"
    >
      <img
        alt="Html"
        src={Object {}}
      />
      Html
      Learn JS For Beginners With Free Book 
      7 months ago
    </div>
  </span>,
  <span
    className="collection-item"
  >
    <div
      className="sc-AxjAm jIgIEX"
    >
      <img
        alt="Html"
        src={Object {}}
      />
      Html
      Learn JS For Beginners With Free Book 
      1 year ago
    </div>
  </span>,
  <span
    className="collection-item"
  >
    <div
      className="sc-AxjAm jIgIEX"
    >
      <img
        alt="Html"
        src={Object {}}
      />
      Html
      Learn JS For Beginners With Free Book 
      1 month ago
    </div>
  </span>,
  <span
    className="collection-item"
  >
    <div
      className="sc-AxjAm jIgIEX"
    >
      <img
        alt="Html"
        src={Object {}}
      />
      Html
      Learn JS For Beginners With Free Book 
      9 years after
    </div>
  </span>,
]
`;
